staging/src/k8s.io/apimachinery/pkg/util/proxy/transport.go

```diff
@@ -101,15 +101,7 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	resp, err := rt.RoundTrip(req)
 
 	if err != nil {
-		message := fmt.Sprintf("Error trying to reach service: '%v'", err.Error())
-		resp = &http.Response{
```

Comment:
I think this is re-used in several places.  I'm not sure about changing it globally.  Perhaps allowing some kind of option for how to handle this or a different transport entirely?